ngraph/core/reference/src/runtime/reference/non_max_suppression.cpp

```diff
@@ -262,7 +262,14 @@ namespace ngraph
                 {
                     std::sort(filteredBoxes.begin(),
                               filteredBoxes.end(),
-                              [](const BoxInfo& l, const BoxInfo& r) { return l.score > r.score; });
+                              [](const BoxInfo& l, const BoxInfo& r) {
+                                  return (l.score > r.score) ||
+                                         (l.score == r.score && l.batch_index < r.batch_index) ||
+                                         (l.score == r.score && l.batch_index == r.batch_index &&
+                                          l.class_index < r.class_index) ||
```

Comment:
Can you compare results of NMS-5 using such boxes comparator with results from ONNX tests?

Reply:
Tests passed.
```
[ RUN      ] INTERPRETER.nonmaxsuppression_center_point_box_format
[       OK ] INTERPRETER.nonmaxsuppression_center_point_box_format (4 ms)
[ RUN      ] INTERPRETER.nonmaxsuppression_flipped_coordinates
[       OK ] INTERPRETER.nonmaxsuppression_flipped_coordinates (1 ms)
[ RUN      ] INTERPRETER.nonmaxsuppression_identical_boxes
[       OK ] INTERPRETER.nonmaxsuppression_identical_boxes (1 ms)
[ RUN      ] INTERPRETER.nonmaxsuppression_limit_output_size
[       OK ] INTERPRETER.nonmaxsuppression_limit_output_size (1 ms)
[ RUN      ] INTERPRETER.nonmaxsuppression_single_box
[       OK ] INTERPRETER.nonmaxsuppression_single_box (0 ms)
[ RUN      ] INTERPRETER.nonmaxsuppression_suppress_by_IOU
[       OK ] INTERPRETER.nonmaxsuppression_suppress_by_IOU (1 ms)
[ RUN      ] INTERPRETER.nonmaxsuppression_suppress_by_IOU_and_scores
[       OK ] INTERPRETER.nonmaxsuppression_suppress_by_IOU_and_scores (1 ms)
[ RUN      ] INTERPRETER.nonmaxsuppression_two_batches
[       OK ] INTERPRETER.nonmaxsuppression_two_batches (1 ms)
[ RUN      ] INTERPRETER.nonmaxsuppression_two_classes
[       OK ] INTERPRETER.nonmaxsuppression_two_classes (1 ms)
[----------] 9 tests from INTERPRETER (11 ms total)
```